### PR TITLE
add test; no duplicate files

### PR DESCRIPTION
**Pull request by BobBorges**
add test

### CODE REVIEW

File: .github/workflows/data-integrity.yml [hunk 1]
```diff
@@ -182,3 +182,22 @@ jobs:
       with:
         github_token: ${{ secrets.GITHUB_TOKEN }}
         branch: ${{ github.head_ref }}
+
+
+  no-duplicate-files:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
```
Comment (ninpnin) [inline .github/workflows/data-integrity.yml]: Let's use Python 3.13 on new workflows

File: test/duplicate-files.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+logger = get_logger("lumberjack")
+
+
+
+
+class DuplicateFilesTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
```
Comment (ninpnin) [inline test/duplicate-files.py:22]: Would it be good to not only check filename duplicates, but also read each file in, calculate a hash, and check that none of these hashes are the same.

This is not strictly necessary but nice to have (?)

**Comment by BobBorges** [comment]
@ninpnin Thx for comments. Done

File: test/duplicate-files.py [hunk 3]
```diff
@@ -27,11 +30,47 @@ def tearDownClass(cls):
...
     def test_no_duplicates(self):
+        """
+        Test no filenames are duplicated
+        """
+        logger.info("Testing there are no duplicate file names...")
         try:
             assert len(self.motions) == len(set(self.motions))
-            logger.info("No duplicate files. 👍👍👍")
+            lower_motions = [f.lower() for f in self.motions]
+            assert len(lower_motions) == len(set(lower_motions))
```
Comment (ninpnin) [inline test/duplicate-files.py]: Use self.assertEqual instead?

File: test/duplicate-files.py [hunk 4]
```diff
@@ -27,11 +30,47 @@ def tearDownClass(cls):
...
+            with open(path, "rb") as f:
+                while chunk := f.read(chunk_size):
+                    h.update(chunk)
+            return h.hexdigest()
+
+        # group files by size
+        size_groups = defaultdict(list)
+        for m in self.motions:
+            size_groups[Path(m).stat().st_size].append(m)
+        # hash files in size groups
```
Comment (ninpnin) [inline test/duplicate-files.py]: Is the grouping necessary?

**Comment by ninpnin** [comment]
Excuse the pedantry, I just want to hammer home https://github.com/swerik-project/the-swedish-parliament-corpus/issues/111

File: test/duplicate-files.py (hunk 4 above)
Comment (BobBorges) [reply test/duplicate-files.py]: faster without grouping :D 

image

...goes down to 4.1s when I remove the dummy error.

File: test/duplicate-files.py (hunk 3 above)
Comment (BobBorges) [reply test/duplicate-files.py]: https://github.com/swerik-project/riksdagen-motions/pull/99/commits/e0557c346087a06d1bf2ca51cc5155d0d9d12779

**Comment by BobBorges** [comment]
@ninpnin pls approve

**Comment by ninpnin** [comment]
@BobBorges I made minor changes, IMO we can merge

EDIT: Does the test actually get run in a Workflow though?
